.github/workflows/release.yml

```diff
@@ -2,86 +2,44 @@ name: Release
 
 on:
   push:
+    branches: [main]
     tags:
       - "v*"
+  pull_request:
+    branches: [main]
 
 jobs:
   build:
     strategy:
+      fail-fast: false
       matrix:
         include:
           - target: x86_64-unknown-linux-gnu
             os: ubuntu-latest
+          - target: aarch64-unknown-linux-gnu
+            os: ubuntu-latest
           - target: x86_64-pc-windows-gnu
             os: ubuntu-latest
             extension: .exe
-          - target: x86_64-apple-darwin
+          # - target: x86_64-apple-darwin
+          #   os: ubuntu-latest
```

Comment:
I think this os should be `macos-latest` though, it seems you've tried both

https://github.com/int128/split-test/actions/runs/8845858159/job/24290599577